kubernetes/samples/scripts/terraform/vcn.tf

```diff
@@ -0,0 +1,414 @@
+resource "oci_core_default_security_list" "oke-default-security-list" {
+  manage_default_resource_id = "${oci_core_virtual_network.oke-vcn.default_security_list_id}"
+  display_name               = "${var.cluster_name}-default-security-list"
+
+  // allow outbound tcp traffic on all ports
+  egress_security_rules {
+    destination = "0.0.0.0/0"
+    protocol    = "all"
+  }
+
+  // allow inbound ssh traffic
+  ingress_security_rules {
+    protocol  = "6"         // tcp
+    source    = "0.0.0.0/0"
+    stateless = false
+
+    tcp_options {
+      "min" = 22
+      "max" = 22
+    }
+  }
+
+  // allow inbound icmp traffic of a specific type
+  ingress_security_rules {
+    protocol = 1
+    source   = "0.0.0.0/0"
+
+    icmp_options {
+      "type" = 3
+      "code" = 4
+    }
+  }
+}
+
+/*
+ * Security list for the worker subnets.
+ *  - Stateless ingress/egress rule-pairs for the worker subnets. this lets traffic between the worker
+ *    nodes flow freely. Stateless rule.
+ *  - Contains a stateful rule to allow traffic to the internet - like for pulling docker images from 
+ *    DockerHub
+ *  - Conatins two ingress rules to allow SSH traffic from OCI Cluster service.
+ */
+resource "oci_core_security_list" "oke-worker-security-list" {
+  compartment_id = "${var.compartment_ocid}"
+  display_name   = "${var.cluster_name}-Workers-SecList"
+  vcn_id         = "${oci_core_virtual_network.oke-vcn.id}"
+
+  egress_security_rules = [
+    {
+      destination = "0.0.0.0/0"
+      protocol    = "6"         // outbound TCP to the internet
+      stateless   = false
+    },
+    {
+      destination = "${var.vcn_cidr_prefix}.10.0/24"
+      protocol    = "all"
+      stateless   = true
+    },
+    {
+      destination = "${var.vcn_cidr_prefix}.11.0/24"
+      protocol    = "all"
+      stateless   = true
+    },
+    {
+      destination = "${var.vcn_cidr_prefix}.12.0/24"
+      protocol    = "all"
+      stateless   = true
+    },
+  ]
+
+  ingress_security_rules = [
+    {
+      # Intra VCN traffic - this lets the 3 subnets in teh 3 ADs tak to each other without restriction.
+      # These are stateless, so they need to be accompanied by stateless egress rules.
+      stateless = true
+
+      protocol = "all"
+      source   = "${var.vcn_cidr_prefix}.10.0/24"
+    },
+    {
+      stateless = true
+      protocol  = "all"
+      source    = "${var.vcn_cidr_prefix}.11.0/24"
+    },
+    {
+      stateless = true
+      protocol  = "all"
+      source    = "${var.vcn_cidr_prefix}.12.0/24"
+    },
+    {
+      # ICMP 
+      protocol = 1
+      source   = "0.0.0.0/0"
+
+      icmp_options {
+        "type" = 3
+        "code" = 4
+      }
+    },
+    {
+      # OCI Cluster service
+      protocol  = "6"             // tcp
+      source    = "130.35.0.0/16"
+      stateless = false
+
+      tcp_options {
+        "min" = 22
+        "max" = 22
+      }
+    },
+    {
+      protocol  = "6"            // tcp
+      source    = "138.1.0.0/17"
```

Comment:
and this?

Reply:
This is also something that was added for OCI services to do management operations on the nodes I believe.

A lot of this goes back to the early days of OKE.  Most likely they should be compared with the latest TF examples/recommendations by the OKE/TF plugin teams.